Tests/src/InternalTests/CommonTypeTests/Map.cpp

```diff
@@ -0,0 +1,193 @@
+CSP_INTERNAL_TEST(CSPEngine, CommonMapTests, MapCopyAssignmentTest)
+{
+	Map<int, String> OldMap;
+	OldMap = {{1, "One"}, {2, "Two"}, {3, "Three"}};
+
+	Map<int, String> MyMap = OldMap;
+
+	// Check if the map contains the initial values
+	EXPECT_TRUE(MyMap.HasKey(1));
+	EXPECT_TRUE(MyMap.HasKey(2));
+	EXPECT_TRUE(MyMap.HasKey(3));
+
+	// Check if the map does not contain an invalid key
+	EXPECT_FALSE(MyMap.HasKey(4));
+
+	EXPECT_EQ(MyMap.Size(), 3);
+}
+
+// Test case to check if elements are initialised with a Map correctly
+CSP_INTERNAL_TEST(CSPEngine, CommonMapTests, MapMoveAssignmentTest)
+{
+	Map<int, String> OldMap;
+	OldMap = {{1, "One"}, {2, "Two"}, {3, "Three"}};
+
+	Map<int, String> MyMap = std::move(OldMap);
+
+	// Check if the map contains the initial values
+	EXPECT_TRUE(MyMap.HasKey(1));
+	EXPECT_TRUE(MyMap.HasKey(2));
+	EXPECT_TRUE(MyMap.HasKey(3));
+
+	// Check if the map does not contain an invalid key
+	EXPECT_FALSE(MyMap.HasKey(4));
+
+	EXPECT_EQ(MyMap.Size(), 3);
+}
+
+// Test case to check if elements are assigned correctly
+CSP_INTERNAL_TEST(CSPEngine, CommonMapTests, MapAssignmentTest)
+{
+	Map<int, String> MyMap;
+	MyMap[1] = "One";
+	MyMap[2] = "Two";
+	MyMap[3] = "Three";
+
+	// Check if the map contains the initial values
+	EXPECT_TRUE(MyMap.HasKey(1));
+	EXPECT_TRUE(MyMap.HasKey(2));
+	EXPECT_TRUE(MyMap.HasKey(3));
+
+	// Check if the map does not contain an invalid key
+	EXPECT_FALSE(MyMap.HasKey(4));
+
+	EXPECT_EQ(MyMap.Size(), 3);
+}
+
+// Test case to check if elements are removed correctly
+CSP_INTERNAL_TEST(CSPEngine, CommonMapTests, MapDeletionTest)
+{
+	Map<int, String> MyMap;
+	MyMap = {{1, "One"}, {2, "Two"}, {3, "Three"}};
+
+	// Remove an element from the map
+	MyMap.Remove(2);
+
+	// Check if the removed element is not in the map
+	EXPECT_FALSE(MyMap.HasKey(2));
+
+	// Check if the other elements are still in the map
+	EXPECT_TRUE(MyMap.HasKey(1));
+	EXPECT_TRUE(MyMap.HasKey(3));
+
+	EXPECT_EQ(MyMap.Size(), 2);
+}
+
+// Test case to check if elements are updated correctly
+CSP_INTERNAL_TEST(CSPEngine, CommonMapTests, MapUpdateTest)
+{
+	Map<int, String> MyMap;
+	MyMap = {{1, "One"}, {2, "Two"}, {3, "Three"}};
+
+	// Update an element in the map
+	MyMap[2] = "NewTwo";
+
+	// Check if the updated element has the correct value
+	EXPECT_EQ(MyMap[2], "NewTwo");
+
+	EXPECT_EQ(MyMap.Size(), 3);
+}
+
+CSP_INTERNAL_TEST(CSPEngine, CommonMapTests, MapKeysValuesClearTest)
+{
+	Map<int, String> MyMap;
+	MyMap = {{1, "One"}, {2, "Two"}, {3, "Three"}};
+
+	EXPECT_EQ(MyMap.Keys()->Size(), 3);
+	EXPECT_FALSE(MyMap.Keys()->Data() == nullptr);
+
+	EXPECT_EQ(MyMap.Values()->Size(), 3);
+	EXPECT_FALSE(MyMap.Values()->Data() == nullptr);
+
+	MyMap.Clear();
+
+	EXPECT_EQ(MyMap.Size(), 0);
+	EXPECT_EQ(MyMap.Keys()->Size(), 0);
+	EXPECT_TRUE(MyMap.Keys()->Data() == nullptr);
+	EXPECT_EQ(MyMap.Values()->Size(), 0);
+	EXPECT_TRUE(MyMap.Values()->Data() == nullptr);
+}
```

Comment:
This should be split into 3 separate tests. We want to keep our unit tests as small as possible and only test a single unit at a time.